src/annotator/delegator.js

```diff
@@ -0,0 +1,95 @@
+import $ from 'jquery';
+
+// Adapted from:
+//  https://github.com/openannotation/annotator/blob/v1.2.x/src/class.coffee
+//
+//  Annotator v1.2.10
+//  https://github.com/openannotation/annotator
+//
+//  Copyright 2015, the Annotator project contributors.
+//  Dual licensed under the MIT and GPLv3 licenses.
+//  https://github.com/openannotation/annotator/blob/master/LICENSE
+
+/**
+ * `Delegator` is a base class for many objects in the annotator.
+ *
+ * It provides:
+ *
+ *  - An event bus, attached to the DOM element passed to the constructor.
+ *    When an event is published, all `Delegator` instances that share the same
+ *    root element will be able to receive the event.
+ *  - Configuration storage (via `this.options`)
+ *  - A mechanism to clean up event listeners and other resources added to
+ *    the page by implementing a `destroy` method, which will be called when
+ *    the annotator is removed from the page.
+ */
+export default class Delegator {
+  /**
+   * Construct the `Delegator` instance.
+   *
+   * @param {HTMLElement} element
+   * @param {Object} [config]
+   */
+  constructor(element, config) {
+    // Some subclasses rely on a legacy mechanism of defining options at
+    // the class level by defining an `options` property on the prototype.
```

Comment:
Will this eventually go away when we migrate/refactor subclasses? (p.s. thanks for this comment; I could see this raising the questioning eyebrows of future devs)

Reply:
Yes. To begin with I expect we'll require subclasses to pass all options in the `config` parameter.